kamu-core/src/infra/ingest/fetch_service.rs

```diff
@@ -529,24 +491,39 @@ impl FetchService {
             listener.on_progress(&progress);
         };
 
-        // TODO: Use symlinks
-        // TODO: Support compression
+        // TODO: PERF: Use symlinks
+        // TODO: PERF: Support compression
         fs_extra::file::copy_with_progress(path, target_path, &options, handle).int_err()?;
 
-        Ok(ExecutionResult {
-            was_up_to_date: false,
-            checkpoint: new_checkpoint,
-        })
+        Ok(FetchResult::Updated(FetchResultUpdated {
+            source_state: Some(PollingSourceState::LastModified(mod_time)),
+            source_event_time: Some(mod_time),
+            has_more: false,
+        }))
     }
 
     fn fetch_http(
         url: &Url,
         headers: &Vec<RequestHeader>,
-        _event_time_source: Option<&EventTimeSource>,
-        old_checkpoint: Option<FetchCheckpoint>,
+        event_time_source: Option<&EventTimeSource>,
+        prev_source_state: Option<&PollingSourceState>,
         target_path: &Path,
         listener: &dyn FetchProgressListener,
-    ) -> Result<ExecutionResult<FetchCheckpoint>, IngestError> {
+    ) -> Result<FetchResult, IngestError> {
+        match event_time_source {
+            None | Some(EventTimeSource::FromMetadata) => (),
+            Some(src) => {
+                return Err(EventTimeSourceError::incompatible(format!(
+                    concat!(
+                        "Url source only supports deriving event time from metadata ",
+                        "(last modified time), found: {:?}",
+                    ),
+                    src
+                ))
```

Comment:
Should we forbid entering `FromPath` in UI ?

Reply:
Eventually we should for usability, but not a priority right now.